lib/util.mli

```diff
@@ -118,6 +118,12 @@ val delayed_force : 'a delayed -> 'a
    Credit X.Leroy, D.Remy. *)
 val try_finally: ('a -> 'b) -> 'a -> ('c -> unit) -> 'c -> 'b
 
+val protect_ref : scope:(unit -> 'a) -> 'b ref -> 'a
```

Comment:
Have a new version that documents all this, will upload after the bench completes.

Reply:
no need to wait

Reply:
:+1:

Ok, will do tomorrow. If we decide to ship this in 8.20 (hopefully) it shouldn't be too much of a PITA as it doesn't need overlays.

Reply:
I mainly wanna be sure the version I push passes all my testing, which takes a little bit.